File: Leet Code/27.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int count=0;
        int i=0,j=nums.size()-1;

        while(i<=j){
            while(i<=j && nums[i]!=val){
                i++;
            }
            if(i>j) break;

            while(i<=j && nums[j]==val){
                j--;
                count++;
            }
            if(i<=j){
                swap(nums[i],nums[j]);
                count++;
                i++;j--;
            }
            
        }
        return nums.size()-count;
    }
};

File: Leet Code/27.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int count=0;
        int i=0;
        int n=nums.length;

        while(i<n){
            if(nums[i]!=val){
                i++;
                count++;
                continue;
            } 
            for(int j=i;j<n;j++){
                if(j>=nums.length-1) break;
                nums[j]=nums[j+1];
            }
            n--;    
        }
        return count;
    }
}
